src/commands/revisionDraft/RevisionDraftFileSystem.ts

```diff
@@ -0,0 +1,197 @@
+import type { Template } from "@azure/arm-appcontainers";
+import { nonNullValueAndProp } from "@microsoft/vscode-azext-utils";
+import { Disposable, Event, EventEmitter, FileChangeEvent, FileChangeType, FileStat, FileSystemProvider, FileType, TextDocument, Uri, window, workspace } from "vscode";
+import { URI } from "vscode-uri";
+import { ext } from "../../extensionVariables";
+import { ContainerAppItem } from "../../tree/ContainerAppItem";
+import type { ContainerAppsItem } from "../../tree/ContainerAppsBranchDataProvider";
+import type { RevisionDraftItem } from "../../tree/revisionManagement/RevisionDraftItem";
+import type { RevisionItem } from "../../tree/revisionManagement/RevisionItem";
+import { localize } from "../../utils/localize";
+
+const notSupported: string = localize('notSupported', 'This operation is not currently supported.');
+
+export class RevisionDraftFile implements FileStat {
+    type: FileType = FileType.File;
+    size: number;
+    ctime: number;
+    mtime: number;
+
+    contents: Uint8Array;
+
+    constructor(contents: Uint8Array, readonly containerAppId: string, readonly baseRevisionName: string) {
+        this.contents = contents;
+        this.size = contents.byteLength;
+        this.ctime = Date.now();
+        this.mtime = Date.now();
+    }
+}
+
+/**
+ * File system provider that allows for reading/writing/deploying container app revision drafts
+ *
+ * Enforces a policy of one revision draft per container app
+ */
+export class RevisionDraftFileSystem implements FileSystemProvider {
+    static readonly scheme: string = 'containerAppsRevisionDraft';
+
+    private readonly emitter: EventEmitter<FileChangeEvent[]> = new EventEmitter<FileChangeEvent[]>();
+    private readonly bufferedEvents: FileChangeEvent[] = [];
+    private fireSoonHandle?: NodeJS.Timer;
+
+    private draftStore: Record<string, RevisionDraftFile> = {};
```

Comment:
It's not necessary, but I think you could simplify some stuff by making this a `Map<string, RevisionDraftFile>` type since it has built in methods for `has` and `delete`,